purchase_ext_sst/models/purchase_order.py

```diff
@@ -119,48 +128,74 @@ def write(self, vals):
                          or False
         if self.is_default_partner(vals['partner_id']):
-            if val_phone:
-                partner_e = self.get_partner_from_phone(val_phone)
+            if val_phone or val_mobile:
+                partner_e = self.get_partner_from_phone(val_phone, val_mobile)
                 if partner_e and vals['partner_id'] != partner_e.id:
                     vals['partner_id'] = partner_e.id
+                    if not vals['phone']:
+                        vals['phone'] = partner_e.phone
+                    if not vals['mobile']:
+                        vals['mobile'] = partner_e.mobile
                 if not partner_e:
-                    vals['partner_id'] = self.create_partner(val_phone).id
+                    vals['partner_id'] = self.create_partner(val_phone,
+                                                             val_mobile).id
                 partner = self.env['res.partner'].browse(vals['partner_id'])
-                self.update_partner(partner, False, val_tent_name)
+                self.update_partner(partner, val_phone, val_mobile,
+                                    val_tent_name)
         else:
-            if val_phone:
-                partner_e = self.get_partner_from_phone(val_phone)
+            if val_phone or val_mobile:
+                partner_e = self.get_partner_from_phone(val_phone, val_mobile)
                 if partner_e and vals['partner_id'] != partner_e.id:
                     vals['partner_id'] = partner_e.id
                 partner = self.env['res.partner'].browse(vals['partner_id'])
-                if val_phone and partner.phone != val_phone:
-                    partner.phone = val_phone
-                self.update_partner(partner, False, val_tent_name)
+                self.update_partner(partner, val_phone, val_mobile,
+                                    val_tent_name)
         return super(PurchaseOrder, self).create(vals)
 
-    def get_partner_from_phone(self, phone):
-        partner = False
-        if phone:
-            partners = self.env['res.partner'].search([
-                '|', ('mobile', '=', phone), ('phone', '=', phone)])
-            partner = partners[0] if partners else False
-        return partner
+    def get_partner_from_phone(self, phone, mobile):
+        Partner = self.env['res.partner']
+        if phone and mobile:
+            # here we use "or" condition instead of "and"
+            partners = Partner.search([
+                '|', ('phone', '=', phone), ('mobile', '=', mobile),
+                ('supplier', '=', True)])
+        elif phone:
+            partners = Partner.search([
+                ('phone', '=', phone), ('supplier', '=', True)])
+        elif mobile:
+            partners = Partner.search([
+                ('mobile', '=', mobile), ('supplier', '=', True)])
+        if partners and len(partners) > 1:
+            raise UserError(_("There is more than one supplier for the "
```

Comment:
Maybe we could do something similar to other modules? e.g.  https://github.com/rfhk/sts-custom/blob/1f0bf55f3ac1ff6cb5b5598099450fec5b9d2bb0/stock_quant_internal_transfer/wizard/stock_quant_transfer.py#L88-L98

```
if partners and len(partners) > 1:
	conflicts_users_list = ''
	for partner in partners:
		conflicts_users_list += _('\n%s\n- Phone: %s\n- '
								  'Mobile: %s\n') % (
			partner.name,
			partner.phone,
			partner.mobile
		)
	raise UserError(_('The entered phone(%s) / mobile(%s) '
					  'conflicts with the following user(s):\n%s') %
					(phone or 'N/A', mobile or 'N/A',
					 conflicts_users_list))
```

Reply:
Would be a nice-to-have feature but not a must as such conflicts are expected to happen very rarely.  Please take over this PR (close this and create a new one) and implement the change as necessary.